src/Workspaces/Core/Portable/Storage/SQLite/v2/SQLitePersistentStorage_ProjectIds.cs

```diff
@@ -20,15 +23,15 @@ internal partial class SQLitePersistentStorage
         /// Given a project, and the name of a stream to read/write, gets the integral DB ID to 
         /// use to find the data inside the ProjectData table.
         /// </summary>
-        private bool TryGetProjectDataId(SqlConnection connection, Project project, string name, out long dataId)
+        private bool TryGetProjectDataId(SqlConnection connection, ProjectKey project, Project? projectOpt, string name, out long dataId)
         {
             dataId = 0;
 
             // First, try to get all the IDs for this project in sync with the DB.
             // This will only be expensive the first time we do this.  But will save
             // us from tons of back-and-forth as any BG analyzer processes all the
             // documents in a solution.
-            BulkPopulateProjectIds(connection, project, fetchStringTable: true);
+            BulkPopulateProjectIds(connection, projectOpt, fetchStringTable: true);
```

Comment:
We always run bulk population on any request that comes in that actually passed a Solution/Proj/Doc.  The way bulk population works tehn is taht it goes to each project and says "have you been bulk populated" and if not, it goes and does that.  So it will still happen at the same time as before.  Just that callers can also call in and have it not happen.

Bulk population is simply an optimization so we can grab a bunch of data immediately from teh DB and put into memory for faster access.